js/source/image_source.js

```diff
@@ -57,19 +56,27 @@ ImageSource.prototype = util.inherit(Evented, {
     onAdd: function(map) {
         this.map = map;
         if (this.image) {
-            this.createTile();
+            this.setCoordinates();
         }
```

Comment:
This method gets fired when the `ImageSource` is added to an instance of `Map`. There are two cases to consider:

 - the HTTP request for the image finishes _before_ the `ImageSource` gets added to the map
 - the HTTP request for the image returns _after_ the `ImageSource` gets added to the map

The code in this method deals with the first case.

I think the lack of an argument to `setCoordinates` is a mistake. The line should be

```js
this.setCoordinates(this.coordinates)
```